src/components/codecSupportHelper.ts

```diff
@@ -102,22 +102,28 @@ export function getMaxBitrateSupport(): number {
 /**
  * Get the max supported video width the active Cast device supports.
  *
- * @param deviceId Cast device id.
  * @returns Max supported width.
  */
-export function getMaxWidthSupport(deviceId: number): number {
-    switch (deviceId) {
-        case deviceIds.ULTRA:
-        case deviceIds.CCGTV:
-            return 3840;
-        case deviceIds.GEN1AND2:
-        case deviceIds.GEN3:
-            return 1920;
-        case deviceIds.NESTHUBANDMAX:
-            return 1280;
-    }
+export function getMaxWidthSupport(): number {
+    // with HLS, it will produce a manifest error if we
+    // send any stream larger than the screen size...
+    return window.innerWidth;
 
-    return 0;
+    // Preferred way to do it:
+    // The devices are capable of decoding and downscaling,
+    // they just refuse to do it with HLS. This would allow
+    // more direct play
+    //switch (deviceId) {
+    //    case deviceIds.ULTRA:
+    //    case deviceIds.CCGTV:
+    //        return 3840;
+    //    case deviceIds.GEN1AND2:
+    //    case deviceIds.GEN3:
+    //        return 1920;
+    //    case deviceIds.NESTHUBANDMAX:
+    //        return 1280;
+    //}
+    //return 0;
```

Comment:
I think we don't want this comment here to be merged?

Reply:
I did not feel good about deleting it. Each method has its benefits and drawbacks.

Using innerWidth is good for HLS because of the strict checking, however if we don't use HLS and don't get this artificial limitation, we can get more directplays if a chromecast wired up to 720p is able to downscale by itself.

It feels wrong to nuke our knowledge about what devices have which limitation.

Ultimately I think both needs to be implemented, one for HLS, one for directplay types